src/main/java/net/neoforged/neoforge/common/conditions/FlagCondition.java

```diff
@@ -5,68 +5,53 @@
+ * Condition checking that a set of {@link FeatureFlag feature flags} are enabled.
  *
  * @apiNote Mainly to be used when flagged content is not contained within the same feature pack which also enables said {@link FeatureFlag feature flags}.
  */
 public final class FlagCondition implements ICondition {
     public static final MapCodec<FlagCondition> CODEC = RecordCodecBuilder.mapCodec(instance -> instance.group(
-            FeatureFlags.CODEC.fieldOf("flags").forGetter(condition -> condition.requiredFeatures),
-            Codec.BOOL.lenientOptionalFieldOf("expected_result", true).forGetter(condition -> condition.expectedResult)).apply(instance, FlagCondition::new));
+            FeatureFlags.CODEC.fieldOf("flags").forGetter(condition -> condition.requiredFeatures)).apply(instance, FlagCondition::new));
 
     private final FeatureFlagSet requiredFeatures;
-    private final boolean expectedResult;
 
-    private FlagCondition(FeatureFlagSet requiredFeatures, boolean expectedResult) {
+    private FlagCondition(FeatureFlagSet requiredFeatures) {
+        if (requiredFeatures.isEmpty()) {
+            throw new IllegalArgumentException("FlagCondition requires a non-empty feature flag set");
+        }
         this.requiredFeatures = requiredFeatures;
-        this.expectedResult = expectedResult;
     }
 
     @Override
     public boolean test(IContext context) {
-        var flagsEnabled = requiredFeatures.isSubsetOf(context.enabledFeatures());
-        // true if: 'expectedResult' is true nd all given flags are enabled
-        // false if: `enabledEnabled' is false and all given flags are disabled
-        return flagsEnabled == expectedResult;
+        return requiredFeatures.isSubsetOf(context.enabledFeatures());
     }
 
     @Override
     public MapCodec<? extends ICondition> codec() {
         return CODEC;
     }
 
-    public static ICondition isEnabled(FeatureFlagSet requiredFeatures) {
-        return new FlagCondition(requiredFeatures, true);
+    public static ICondition of(FeatureFlagSet requiredFeatures) {
+        return new FlagCondition(requiredFeatures);
     }
 
-    public static ICondition isEnabled(FeatureFlag requiredFlag) {
-        return isEnabled(FeatureFlagSet.of(requiredFlag));
-    }
-
-    public static ICondition isEnabled(FeatureFlag requiredFlag, FeatureFlag... requiredFlags) {
-        return isEnabled(FeatureFlagSet.of(requiredFlag, requiredFlags));
-    }
-
-    public static ICondition isDisabled(FeatureFlagSet requiredFeatures) {
-        return new FlagCondition(requiredFeatures, false);
-    }
-
-    public static ICondition isDisabled(FeatureFlag requiredFlag) {
-        return isDisabled(FeatureFlagSet.of(requiredFlag));
-    }
-
-    public static ICondition isDisabled(FeatureFlag requiredFlag, FeatureFlag... requiredFlags) {
-        return isDisabled(FeatureFlagSet.of(requiredFlag, requiredFlags));
```

Comment:
Please retain these methods, I find them much better to read and understand then `new NotCondition(FlagCondition.of(...))` and given that they return ICondition, the actual implementation does not matter

Reply:
Creating the conditions manually is not the recommended way of creating them, using the builder is. The builder is simply `not(...)`

Reply:
even going that route `not(featureFlags())` doesnt really explain what it does by simply reading the methods names, where as `not(featureFlagsEnabled())` or `not(isFeatureEnabled())` does.

Reply:
I do agree with renaming the methods. However we don't have modIsntLoaded or itemDoesntExist or tagIsntEmpty or andNot or orNot utilities. Conditions are boolean composition and we would be bloating the api surface by introducing combinations instead of having people compose the condition exactly as it would be output in the json.